composer/metrics/nlp.py

```diff
@@ -270,19 +326,31 @@ def update(self, outputs: List[str], labels: List[List[str]], batch: Optional[Di
         if batch is None:
             batch = {}
         cot_delimiter = batch.get('cot_delimiter', '')
-        for sample_output, sample_labels in zip(outputs, labels):
+        for sample_output, sample_labels, prompt_tensor in zip(outputs, labels, batch['input_ids']):
             final_answer = sample_output
             if cot_delimiter is not None and len(cot_delimiter) > 0:
                 final_answer = final_answer.split(cot_delimiter)[-1]
 
             cleaned_final_answer = self.normalize_answer(final_answer)
             cleaned_sample_labels = {self.normalize_answer(label) for label in sample_labels}
-
+            correct = False
             if any(cleaned_final_answer.startswith(label) for label in cleaned_sample_labels):
                 self.correct += torch.tensor(1.0)
+                correct = True
+
+            assert isinstance(self.response_cache, list)
```

Comment:
Add `if self.cache_responses` here?

Reply:
response_cache will be a list regardless since it is part of the super class's ctor